.devcontainer/Dockerfile

```diff
@@ -0,0 +1,50 @@
+#-------------------------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See https://go.microsoft.com/fwlink/?linkid=2090316 for license information.
+#-------------------------------------------------------------------------------------------------------------
+
+FROM debian:9
+
+# Avoid warnings by switching to noninteractive
+ENV DEBIAN_FRONTEND=noninteractive
+
+# This Dockerfile adds a non-root 'vscode' user with sudo access. However, for Linux,
+# this user's GID/UID must match your local user UID/GID to avoid permission issues
+# with bind mounts. Update USER_UID / USER_GID if yours is not 1000. See
+# https://aka.ms/vscode-remote/containers/non-root-user for details.
+ARG USERNAME=vscode
+ARG USER_UID=1000
+ARG USER_GID=$USER_UID
+
+# Configure apt and install packages
+RUN apt-get update \
+    && apt-get -y install --no-install-recommends apt-utils dialog 2>&1 \
+    # 
+    # Verify git, process tools, lsb-release (useful for CLI installs) installed
+    && apt-get -y install git iproute2 procps lsb-release \
+    #
+    # Install C++ tools
+    && apt-get -y install build-essential cppcheck valgrind \
+    #
+    ## Install newer version of CMake (OpenJPH requires 3.10+, debian has older version)
+    && apt-get install wget \
+    && wget https://github.com/Kitware/CMake/releases/download/v3.15.2/cmake-3.15.2.tar.gz \
+    && tar -zxvf cmake-3.15.2.tar.gz \
+    && (cd /cmake-3.15.2; ./bootstrap; make -j12 install)\
```

Comment:
-j12 is rather aggressive; you must have a machine with 12 or more cores, and a fast hard drive.
I will reduce it to a more modest -j2.